app/design-system/design-system-hedvig/src/main/kotlin/com/hedvig/android/design/system/hedvig/HedvigBottomSheet.kt

```diff
@@ -25,38 +56,111 @@ import eu.wewox.modalsheet.ModalSheet
+  bottomButtonText: String,
   onSystemBack: (() -> Unit)? = { onVisibleChange(false) },
+  onBottomButtonClick: () -> Unit = { onVisibleChange(false) },
   sheetPadding: PaddingValues? = null,
   cancelable: Boolean = true,
   content: @Composable ColumnScope.() -> Unit,
 ) {
+  val scrollState = rememberScrollState()
+  var scrollDown by remember { mutableStateOf(false) }
+  LaunchedEffect(scrollDown) {
+    if (scrollDown) {
+      scrollState.animateScrollTo(scrollState.maxValue)
+    }
+  }
   val isImeVisible = WindowInsets.isImeVisible
   val defaultPadding = if (isImeVisible) {
     WindowInsets.ime.asPaddingValues()
   } else {
     WindowInsets.safeDrawing
-      .only(WindowInsetsSides.Bottom).asPaddingValues()
+      .only(WindowInsetsSides.Bottom + WindowInsetsSides.Top).asPaddingValues()
   }
   val finalSheetPadding = sheetPadding ?: defaultPadding
+
   ModalSheet(
     visible = isVisible,
     onVisibleChange = onVisibleChange,
     cancelable = cancelable,
-    content = content,
     scrimColor = bottomSheetColors.scrimColor,
     backgroundColor = bottomSheetColors.bottomSheetBackgroundColor,
     contentColor = bottomSheetColors.contentColor,
     sheetPadding = finalSheetPadding,
     onSystemBack = onSystemBack,
     shape = bottomSheetShape.shape,
+  ) {
+    Box {
+      Column(
+        modifier = Modifier.padding(horizontal = bottomSheetShape.contentHorizontalPadding)
+          .verticalScroll(scrollState),
+      ) {
+        Spacer(modifier = Modifier.height(8.dp))
+        LittleUpperChip()
+        Spacer(modifier = Modifier.height(20.dp))
+        content()
+        HedvigButton(
+          onClick = onBottomButtonClick,
+          text = bottomButtonText,
+          enabled = true,
+          buttonStyle = Ghost,
+          modifier = Modifier.fillMaxWidth(),
+        )
+        Spacer(modifier = Modifier.height(32.dp))
+      }
+      if (scrollState.canScrollForward && !scrollDown) {
+        HintArrowDown {
+          scrollDown = true
+        }
+      }
+    }
+  }
+}
+
+@Composable
+private fun ColumnScope.LittleUpperChip() {
+  Surface(
+    modifier = Modifier
+      .width(40.dp)
+      .height(4.dp)
+      .align(Alignment.CenterHorizontally)
+      .background(
+        shape = ShapeDefaults.CornerSmall,
+        color = bottomSheetColors.chipColor,
+      ).clip(ShapeDefaults.CornerSmall),
+  ) {}
+}
+
+@Composable
+private fun BoxScope.HintArrowDown(onClick: () -> Unit) {
+  val infiniteTransition = rememberInfiniteTransition(label = "arrowDownAlpha")
+  val arrowAlpha by infiniteTransition.animateFloat(
+    initialValue = 1f,
+    targetValue = 0.0f,
+    animationSpec = infiniteRepeatable(
+      animation = tween(
+        durationMillis = 2000,
+        easing = LinearEasing,
+      ),
+      repeatMode = RepeatMode.Reverse,
+    ),
+    label = "arrowDownAlpha animation",
   )
+  Row(Modifier.align(Alignment.BottomEnd).padding(horizontal = 32.dp, vertical = 16.dp)) {
+    Icon(
```

Comment:
I have a feeling that I'd like for such an arrow to also have a little bit of a circle background behind it to be a bit more visible.
Could we take a look at how other apps do this perhaps?
At minimum I'd just add a circle behind it that is the opposite color
Also probably good to make it a
```
IconButton(onClick = onClick) {
  Icon()
}
```
To have it be the right touch size for sure.